pallene/driver.lua

```diff
@@ -133,33 +131,49 @@ function driver.compile(argv0, input_ext, output_ext, input_file_name)
 
     local mod_name = string.gsub(base_name, "/", "_")
 
-    local first_step = step_index[input_ext]  or error("invalid extension")
-    local last_step  = step_index[output_ext] or error("invalid extension")
-    assert(first_step < last_step, "impossible order")
+    local ok, errs = false, nil
+    if output_ext == "lua" then
+        assert(input_ext == "pln")
 
-    local file_names = {}
-    for i = first_step, last_step do
-        local step = compiler_steps[i]
-        if (i == first_step or i == last_step) then
-            file_names[i] = base_name .. "." .. step.name
+        local input, err = driver.load_input(input_file_name)
+        if err then
+            errs = { err }
```

Comment:
I think the basic ideas we used in compile_to_c also apply to this function. Try declaring variables like `ok` and `errs` closer to where we need them and see if you can use `return false, errs` or `return false, {err}` instead of nested `else` statements.